build/webcomponents/js/field-category/field-category.js

```diff
@@ -0,0 +1,35 @@
+customElements.define('joomla-field-category', class extends HTMLElement {
+  constructor() {
+    super();
+
+    this.element = '';
+    Joomla.loadingLayer('load', document.body);
+  }
+
+  connectedCallback() {
+    // Check if custom fields are enabled
+    if (this.getAttribute('custom-fields-enabled') !== 'true') {
+      return;
+    }
+
+    this.element = this.querySelector('select');
+    this.categoryHasChanged = this.categoryHasChanged.bind(this);
+
+    if (!this.element.value !== this.getAttribute('custom-fields-cat-id')) {
+      this.element.value = this.getAttribute('custom-fields-cat-id');
+    }
+
+    this.element.addEventListener('change', this.categoryHasChanged);
+  }
+
+  categoryHasChanged() {
+    if (this.element.value === parseInt(this.element.parentNode.getAttribute('custom-fields-cat-id'), 10)) {
+      return;
+    }
+
+    Joomla.loadingLayer('show', document.body);
+
+    document.querySelector('input[name=task]').value = `${this.element.parentNode.getAttribute('custom-fields-section')}.reload`;
```

Comment:
According to https://github.com/joomla-projects/joomla-es6/pull/75/files#diff-4d276fa5ac7a065127d4620092ed38fdR330, `custom-fields-section` is an attribute of the ce. So why do we need to get it from parentNode?

Reply:
Haha this.element.parentNode === this, me drunk 😵